pkg/server/container_create.go

```diff
@@ -634,3 +645,19 @@ func generateUserString(username string, uid, gid *runtime.Int64Value) (string,
 	}
 	return userstr, nil
 }
+
+// generateOCIHooks generates a runtimespec.Hooks struct from json in the passed in filepath.
+func (c *criService) generateOCIHooks(profile string) (*runtimespec.Hooks, error) {
+	if profile == "" {
+		return nil, nil
+	}
+	hooks := &runtimespec.Hooks{}
+	f, err := ioutil.ReadFile(profile)
```

Comment:
We could set a reasonable file size limit to avoid memory exhaustion, especially with user input.

I thought there was a previous Kubernetes security recommendation about this code pattern (ReadFile + Unmarshal) but I can't find it anymore.

Reply:
const reasonable_filesize = initial_reasonable_filesize^(current_year - 1970)